File: app/src/main/java/org/amoustakos/boilerplate/injection/component/ActivityComponent.java
package org.amoustakos.boilerplate.injection.component;


import android.app.Activity;

import org.amoustakos.boilerplate.examples.ui.activities.MainActivity;
import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext;
import org.amoustakos.boilerplate.injection.annotations.scopes.PerActivity;
import org.amoustakos.boilerplate.injection.module.ActivityModule;
import org.amoustakos.boilerplate.ui.activities.BaseActivity;

import dagger.Subcomponent;

/**
 * This component inject dependencies to all Activities across the application
 */
@PerActivity
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {


    // =========================================================================================
    // Injections
    // =========================================================================================

    void inject(Activity activity);
    void inject(MainActivity activity);


    // =========================================================================================
    // Base
    // =========================================================================================

    @ActivityContext BaseActivity activity();

}

File: app/src/main/java/org/amoustakos/boilerplate/injection/component/ActivityComponent.kt
package org.amoustakos.boilerplate.injection.component


import android.app.Activity
import dagger.Subcomponent
import org.amoustakos.boilerplate.examples.ui.activities.ListingActivity
import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext
import org.amoustakos.boilerplate.injection.annotations.scopes.PerActivity
import org.amoustakos.boilerplate.injection.module.injectors.ActivityModule
import org.amoustakos.boilerplate.ui.activities.BaseActivity

/**
 * This component inject dependencies to all Activities across the application
 */
@PerActivity
@Subcomponent(modules = [
	ActivityModule::class
])
interface ActivityComponent {


	// =========================================================================================
	// Injections
	// =========================================================================================

	fun inject(activity: Activity)
	fun inject(activity: ListingActivity)


	// =========================================================================================
	// Base
	// =========================================================================================

	@ActivityContext
	fun activity(): BaseActivity

}
